likecoin/assets/js/likecoin.js

```diff
@@ -11,65 +16,95 @@ function show(selector) {
   elem.style.display = '';
 }
 
-async function likecoinInit() {
+function hide(selector) {
+  const elem = document.querySelector(`.likecoin${selector}`);
+  elem.style.display = 'none';
+}
+
+
+function showError(selector) {
+  webThreeError = selector;
+  const elems = document.querySelectorAll(`.likecoin.webThreeError`);
+  elems.forEach((elem) => elem.style.display = 'none');
+  show(selector);
+}
+
+
+async function pollForWebThree() {
   if (!window.web3) {
-    show('.needMetaMask');
+    showError('.needMetaMask');
+    console.error('no web3');
     return;
   }
-  const {
-    networkVersion,
-    selectedAddress,
-  } = web3.currentProvider.publicConfigStore._state;
-  if (networkVersion !== '1') {
-    show('.needMainNet');
+  webThreeInstance = new Web3(window.web3.currentProvider);
+  const network = await webThreeInstance.eth.net.getNetworkType();
+  if (network !== 'main') {
+    showError('.needMainNet');
+    console.error('not mainnet');
     return;
-  } else if (!selectedAddress) {
-    show('.needUnlock');
+  }
+  const accounts = await webThreeInstance.eth.getAccounts();
+  if (!accounts || !accounts[0]) {
+    showError('.needUnlock');
+    console.error('not unlocked');
     return;
   }
+  const selectedAddress = accounts[0];
+  webThreeError = null;
+  return webThreeInstance.utils.toChecksumAddress(selectedAddress);
+}
+
+async function login() {
+  if (webThreeError) return;
+  let res = await fetch(`${challengeUrl}?wallet=${address}`);
+  const { challenge } = await res.json();
+  const signature = await webThreeInstance.eth.personal.sign(challenge, address);
+  if (err || !signature) {
+    throw (err || 'No signature');
+  }
+  const body = JSON.stringify({ challenge, signature, wallet: address });
+  res = await fetch(challengeUrl, {
+    body,
+    headers: {
+      'content-type': 'application/json',
+    },
+    method: 'POST'
+  });
+  const { user, wallet } = await res.json();
+  handleUpdateId(user, wallet);
+  if (user) {
```

Comment:
So its possible to have `user` undefined? Then what happens for `handleUpdateId(user, wallet);`?